cli/tests/integration/test_infra/test_cockpit.py

```diff
@@ -0,0 +1,129 @@
+import uuid
+from typing import Generator
+
+import pytest
+import requests
+from requests.auth import HTTPBasicAuth
+from scaleway import Client, ScalewayException
+from scaleway.account import v2 as account
+from scaleway.cockpit import v1beta1 as sdk
+
+import cli.infra.cockpit as cpt
+
+
+@pytest.fixture(scope="module")
+def api() -> Generator[sdk.CockpitV1Beta1API, None, None]:
+    """Return the project."""
+    client = Client.from_config_file_and_env()
+    api = account.AccountV2API(client)
+    project = None
+    try:
+        project = api.create_project(
+            name=f"test-cockpit-{uuid.uuid4()}",
+            description="Generated by gateway integration tests",
+        )
+        client.default_project_id = project.id
+        cockpit_api = sdk.CockpitV1Beta1API(client)
+        yield cockpit_api
+    finally:
+        if project:
+            # We need to delete cockpit if it was created
+            cockpit_api = sdk.CockpitV1Beta1API(client)
+            cockpit_api.deactivate_cockpit(project_id=project.id)
+            try:
+                cockpit_api.wait_for_cockpit(project_id=project.id)
+            except ScalewayException as err:
+                if not err.status_code == 404:
+                    raise
+            api.delete_project(project_id=project.id)
+
+
+@pytest.mark.deployed
+def test_ensure_cockpit_activated(api: sdk.CockpitV1Beta1API):
```

Comment:
Can you make the tests in this file part of a single class? This has a few of benefits:

- Fits with the style of the other integration tests
- We only need to tag the class with `@pytest.mark.deployed` (not every method)
- We can share code, e.g. `cpt.ensure_cockpit_activated(api=api)` and `cockpit = api.get_cockpit()`

Have a look at the other tests for how the tests go in a single class

Reply:
Refactored the tests a little bit to be more consistent with the rest.